contributors/design-proposals/storage/container-storage-interface.md

```diff
@@ -410,35 +410,39 @@ When the controller decides to detach a CSI volume, it will call the in-tree CSI
 
 ### Recommended Mechanism for Deploying CSI Drivers on Kubernetes
 
-Although, Kubernetes does not dictate the packaging for a CSI volume driver, it offers the following recommendations to simplify deployment of a containerized CSI volume driver on Kubernetes.
+Although Kubernetes does not dictate the packaging for a CSI volume driver, it offers the following recommendations to simplify deployment of a containerized CSI volume driver on Kubernetes.
 
 ![Recommended CSI Deployment Diagram](container-storage-interface_diagram1.png?raw=true "Recommended CSI Deployment Diagram")
 
 To deploy a containerized third-party CSI volume driver, it is recommended that storage vendors:
 
   * Create a “CSI volume driver” container that implements the volume plugin behavior and exposes a gRPC interface via a unix domain socket, as defined in the CSI spec (including Controller, Node, and Identity services).
-  * Bundle the “CSI volume driver” container with helper containers (external-attacher, external-provisioner, Kubernetes CSI Helper) that the Kubernetes team will provide (these helper containers will assist the “CSI volume driver” container in interacting with the Kubernetes system). More specifically, create the following Kubernetes objects:
-    * A `StatefulSet` (to facilitate communication with the Kubernetes controllers) that has:
-      * Replica size 1
-        * Guarantees that no more than 1 instance of the pod will be running at once (so we don’t have to worry about multiple instances of the `external-provisioner` or `external-attacher` in the cluster).
+  * Bundle the “CSI volume driver” container with helper containers (external-attacher, external-provisioner, node-driver-registrar, cluster-driver-registrar, external-resizer, external-snapshotter, livenessprobe) that the Kubernetes team will provide (these helper containers will assist the “CSI volume driver” container in interacting with the Kubernetes system). More specifically, create the following Kubernetes objects:
+    * To facilitate communication with the Kubernetes controllers, a `StatefulSet` or a `Deployment` (depending on the user's need; see [Cluster-Level Deployment](#cluster-level-deployment)) that has:
       * The following containers
         * The “CSI volume driver” container created by the storage vendor.
-        * The `external-attacher` container provided by the Kubernetes team.
-        * The `external-provisioner` container provided by the Kubernetes team.
+        * Containers provided by the Kubernetes team:
```

Comment:
Theoretically all of these controller sidecars are optional if your driver doesn't need the behavior it provides.